app/controllers/admin/claim_checks_controller.rb

```diff
@@ -34,6 +37,12 @@ def reject_missing_payroll_gender
     end
   end
 
+  def reject_inconsistent_claims
```

Comment:
This reads to me as if we are rejecting the inconsistent claims, rather than rejecting if there are inconsistent claims. Maybe something like `reject_if_inconsistent_claims`?

Reply:
In fact, there now seem to be quite a few validations in the before filters now. How about combining them all into one? Something like:

```ruby
before_action :validate_check

...

def validate_check
  reject_checked_claims
  reject_missing_payroll_gender
  reject_if_inconsistent_claims
end
```

Reply:
`reject_if_inconsistent_claims` is a better name, I'll change that.

As for combining the multiple filters into a single filter, I think that this would modify the behaviour of this controller in a way that we don't want, because we'd lose Rails's "end filter chain if a filter renders or redirects" behaviour. Consider, for example, if we have a claim that has already been rejected because its payroll gender is missing. At the moment, the `reject_checked_claims` filter would redirect, and the filter chain would end. If we combine all the filters into a single method, then we'd end up doing a redirect in `reject_checked_claims`, and then a _second_ redirect in `reject_missing_payroll_gender`, which isn't allowed.

Reply:
We could do `redirect ... and return` which would work though right?

Reply:
No, that would just return from that method, it wouldn't break the filter chain.

Reply:
Oh yeah, I remember trying something similar and being massively confused 😆